### DISCUSSION TITLE
Help with creating library for Ufactory Lite6

### DISCUSSION THREAD

**Discussion by romtecmax** [visose/Robots#214] [Q&A] [answered]
Hi there

I'm in the process of creating a library entry for the Ufactory Lite6 [(Documentation here)](https://www.ufactory.cc/download-lite6-robot).

I created the library file with the robot meshes inside. I used their STEP model, oriented it correctly in space and converting it to single meshes per layer. it appears in the grasshopper preview when I load it into a robot def.

However, I can't seem to get the xml file to work - I entered the values from the Ufactory Docs ( [found here](https://www.ufactory.cc/_files/ugd/896670_244de00e080c4336a28e00ac0c77202b.pdf) ), but the rotations and joints are totally off, except for joint 0 which works fine and joint 1 which, altough the angles areoffset by -90deg, at least revolves around the correct axis. All other joints however, rotate around wrong axes.

I would be very grateful if someone could take a look at this and tell me where my mistake is.

Best
Max

PS: I had to use Manufacturer: UR otherwise I got the error that ufactory is not supported or something like that
[xml & 3dm for library - romtec-lite6.zip](https://github.com/visose/Robots/files/10520633/romtec-lite6.zip)
[my rhino and gh file i used to test - gh setup.zip](https://github.com/visose/Robots/files/10520638/gh.setup.zip)

**Comment by visose** [comment, marked as answer]
Hi @romtecmax, as the error says, UFactory robots are not currently supported. This simulation would only work if their kinematics is identical to a UR robot. But the generated program will still be for UR robots.

If you are interested in sponsoring the work required to add support to this robot let me know and I can send you a quote. The existing support for Staubli, Doosan, Franka Emika have been sponsored in this way.

**Reply by romtecmax** [reply]
Thanks for your reply. Yes I'm aware the program would be for the UR but I was planning to figure out how to translate that into a Ufactory program after the setup.

What do you mean exactly with kinematics being the same - excuse my question please, I'm more on the creative side of things :)

I don't think I'm able to sponsor it, don't even have budget for a proper robot ;) we're not talking three figures, right?

Best,
Max

**Reply by visose** [reply]
Try setting it as an ABB, KUKA, Staubli or Doosan, those use a different kinematics solver than UR. But I would read up on the basics of industrial robots before you attempt this, as it might be more complex than you think, it will be hard to power through via only trial and error.

**Reply by romtecmax** [reply]
Alright, thank you for the help. I'll give it a try and let's see how far i get.
